sdks/python/apache_beam/ml/gcp/cloud_dlp.py

```diff
@@ -0,0 +1,224 @@
+      raise ValueError(
+          'inspection_template_name or inspection_config must be specified')
+    elif inspection_template_name is not None:
+      self.config['inspect_template_name'] = inspection_template_name
+    elif inspection_config is not None:
+      self.config['inspect_config'] = inspection_config
+
+  def expand(self, pcoll):
+    return (
+        pcoll
+        | beam.ParDo(_DeidentifyFn(self.config, self.timeout, self.project)))
+
+
+@experimental()
+class InspectForDetails(beam.PTransform):
+  """Inspects input text for sensitive information.
+  the ``PTransform`` returns a ``PCollection`` of
+  ``List[google.cloud.dlp_v2.proto.dlp_pb2.Finding]``
+  Example usage::
+      pipeline | InspectForDetails(project='example-gcp-project',
+                inspection_config={'info_types': [{'name': 'EMAIL_ADDRESS'}]})
+  """
+  def __init__(
+      self,
+      inspection_template_name=None,
+      inspection_config=None,
+      project=None,
+      timeout=None):
+    """Initializes a :class:`InspectForDetails` transform.
+    Args:
+      inspection_template_name (str): This or `inspection_config` required.
+        Name of inspection template to be used
+        to detect sensitive data in text.
+      inspection_config
+        (``Union[dict, google.cloud.dlp_v2.types.InspectConfig]``):
+        Configuration for the inspector used to detect sensitive data in text.
+      project (str): Required. Name of GCP project in which the processing
+        will take place.
+      timeout (float): Optional. The amount of time, in seconds, to wait for
+        the request to complete.
+    """
+    self.project = project
+    self.timeout = timeout
+    self.config = {}
+    if project is None:
+      raise ValueError(
+          'GCP project name needs to be specified in "project" property')
+    if inspection_template_name is not None and inspection_config is not None:
+      raise ValueError(
+          'Both inspection_template_name and '
+          'inspection_template were specified.'
+          ' Please specify ony one of these.')
+    elif inspection_config is None and inspection_template_name is None:
+      raise ValueError(
+          'inspection_template_name or inspection_config must be specified')
+    elif inspection_template_name is not None:
+      self.config['inspect_template_name'] = inspection_template_name
+    elif inspection_config is not None:
+      self.config['inspect_config'] = inspection_config
+
+  def expand(self, pcoll):
+    return pcoll | beam.ParDo(
+        _InspectFn(self.config, self.timeout, self.project))
+
+
+class _DeidentifyFn(beam.DoFn):
+  def __init__(self, config=None, timeout=None, project=None, client=None):
+    self.config = config
+    self.timeout = timeout
+    self.client = client
+    self.project = project
+
+  def start_bundle(self):
+    if self.client is None:
+      self.client = dlp_v2.DlpServiceClient()
+
+  def process(self, element, **kwargs):
+    params = {
```

Comment:
params and params.update, could it move to start_bundle?

Reply:
I moved it to `setup()`. Will that be OK?